test/puppetlabs/trapperkeeper/testutils/logging_test.clj

```diff
@@ -0,0 +1,150 @@
+(ns puppetlabs.trapperkeeper.testutils.logging-test
+  (:require
+   [clojure.test :refer :all]
+   [clojure.tools.logging :as log]
+   [puppetlabs.kitchensink.core :as kitchensink]
+   [puppetlabs.trapperkeeper.logging :refer [reset-logging root-logger-name]]
+   [puppetlabs.trapperkeeper.testutils.logging :as tgt :refer [event->map]])
+  (:import
+   [org.slf4j Logger LoggerFactory]))
+
+;; Without this, "lein test NAMESPACE" and :only invocations may fail.
+(use-fixtures :once (fn [f] (reset-logging) (f)))
+
+(deftest with-log-level-and-logging-to-atom
+  (let [expected {:logger "puppetlabs.trapperkeeper.testutils.logging-test"
+                  :level :info
+                  :message "wlta-test"
+                  :exception nil}]
+    (let [log (atom [])]
+      (tgt/with-log-level root-logger-name :error
+        (tgt/with-logging-to-atom root-logger-name log
+          (log/info "wlta-test"))
+        (is (not-any? #(= expected %) (map event->map @log)))))
+    (let [log (atom [])]
+      (tgt/with-log-level root-logger-name :info
+        (tgt/with-logging-to-atom root-logger-name log
+          (log/info "wlta-test"))
+        (is (some #(= expected %) (map event->map @log)))))))
+
+(def find-logger #'puppetlabs.trapperkeeper.testutils.logging/find-logger)
+(def log-event-listener #'puppetlabs.trapperkeeper.testutils.logging/log-event-listener)
```

Comment:
Since I never wrote Clojure before PL I don't always feel like I know what the best practices are, but, my knee-jerk reaction to this pattern is concern that it would prevent things like automatic refactorings in dev environments from catching these references.  So we've just erred on the side of not making things private.  If we consider them internal-only then we might do something like suffix the function name with an asterisk or move it to a separate namespace called `internal` just to avoid confusion about what should be considered public API.  Personally I prefer those approaches, but I don't feel strongly enough about it to where if some folks who had more Clojure experience than myself said "this is a normal way to do things" that I'd resist too heavily.